azext_iot/central/models/template.py

```diff
@@ -15,30 +15,78 @@ def __init__(self, template: dict):
             self.name = template.get("displayName")
             self.interfaces = self._extract_interfaces(template)
             self.schema_names = self._extract_schema_names(self.interfaces)
+            self.components = self._extract_components(template)
+            if self.components:
+                self.component_schema_names = self._extract_schema_names(
+                    self.components
+                )
+
         except:
             raise CLIError("Could not parse iot central device template.")
 
-    def get_schema(self, name, interface_name=""):
-        # interface_name has been specified, do a pointed lookup
-        if interface_name:
-            interface = self.interfaces.get(interface_name, {})
-            return interface.get(name)
+    def get_schema(self, name, is_component=False, identifier="") -> dict:
+        entities = self.components if is_component else self.interfaces
+        if identifier:
+            # identifier specified, do a pointed lookup
+            entry = entities.get(identifier, {})
+            return entry.get(name)
 
-        # find first matching name in any interface
-        for interface in self.interfaces.values():
-            schema = interface.get(name)
+        # find first matching name in any component
+        for entry in entities.values():
+            schema = entry.get(name)
             if schema:
                 return schema
 
         # not found
         return None
 
+    def _extract_components(self, template: dict) -> dict:
+        try:
+            dcm = template.get("capabilityModel", {})
+            if dcm.get("contents"):
```

Comment:
can simplify here - 

```
# doing this is safe without an if check because you are defaulting to an empty dictionary
rootContents = dcm.get("contents", {}) 
components = [
                    entity
                    for entity in rootContents
                    if entity.get("@type") == "Component"
]
```

Reply:
fixed